### ISSUE TITLE
Pontoon should use light theme for visitors requesting light theme

### ISSUE THREAD

**Issue by flodolo** [mozilla/pontoon#3339]
On a new profile, set Firefox to request light color scheme and visit Pontoon. Pontoon keeps using the dark theme, while it should follow the user's preferences.

**Comment by mathjazz** [comment]
Is this issue about changing the default theme for newly registered users from the [dark theme](https://github.com/mozilla/pontoon/blob/main/pontoon/base/models/user_profile.py#L34-L38) to the system theme?

What about users that are not authenticated?

**Comment by flodolo** [comment]
No, this is about visitors (i.e. unauthenticad users), and matching their preferences. If the preference is not set, then we would use the default (dark).

**Comment by mathjazz** [comment]
I think the behavior should be the same for unauthenticated users and new users. If you choose a light theme in your system settings and we honor that in Pontoon, why should we switch to the dark theme after you create an account?

Also note that we'd use [prefers-color-scheme](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) to support this behavior, which always returns either "light" or "dark". It's based on the OS and browser settings.

**Comment by flodolo** [comment]
> I think the behavior should be the same for unauthenticated users and new users. If you choose a light theme in your system settings and we honor that in Pontoon, why should we switch to the dark theme after you create an account?

It's a good point. The only reason not to change that default would be that we have introduced theme selection ~15 years into the life of Pontoon, so this has the potential of being disruptive for a lot of active users. 

Consistency or potential annoyance (and added communication to mitigate it)? Not sure which one is better.

**Comment by mathjazz** [comment]
For existing users everything stays the same. That [link above](https://github.com/mozilla/pontoon/issues/3339#issuecomment-2352701710) indicates that we have set the theme to "dark" for all users when the feature was shipped.

The only thing that would potentially impact the existing users is changing the default for unauthenticated users to the "system" theme.

**Comment by flodolo** [comment]
I feel like there's a disconnect somewhere.

* Unauthenticated users: follow user preferences (change from current behavior, where dark is default).
* New users: dark theme.

I read [your comment](https://github.com/mozilla/pontoon/issues/3339#issuecomment-2352753855) as "we should change the default for new users from `Dark theme` to `Match system`. That sounds reasonable, because they might end up registering on a light theme (their OS/Browser choice), and it would become dark after logging in. 

To confirm, would such change affect only new users registering after the change, or also existing users who never set their theme preferences?

**Comment by mathjazz** [comment]
> To confirm, would such change affect only new users registering after the change, or also existing users who never set their theme preferences?

Changing the default for registered users only impacts new users. All existing users already have the theme set to "Dark", "Light" or "System". There is no such thing as "Theme preference not set".

Changing the default for unauthenticated users OTOH impacts all users (when they aren't logged in).

**Comment by flodolo** [comment]
> Changing the default for unauthenticated users OTOH impacts all users (when they aren't logged in).

It changes the current behavior for users who have not set their browser or OS to request dark content.

I think that would that be acceptable? Or do you think the majority of users have light settings, and it would be disruptive?

**Comment by mathjazz** [comment]
I think the default must be the same for unauthenticated users and new accounts.

Not sure what's the best default though - keep doing what we've been doing forever and nobody complained, or start doing what everybody else is doing.
